src/components/ColorSwatch/ColorSwatch.tsx

```diff
@@ -26,13 +27,22 @@ export const ColorSwatch: React.FC<Props> = ({
   soldOut,
   onSale,
   onClick,
-  onHover,
+  handleMouseOver,
+  handleMouseOut,
   ...rest
 }) => {
   const handleClick = useCallback((_event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     onClick && onClick(color);
   }, []);
 
+  const onMouseOver = useCallback((_event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    handleMouseOver && handleMouseOver(color);
+  }, []);
+
+  const onMouseOut = useCallback((_event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    handleMouseOut && handleMouseOut();
+  }, []);
```

Comment:
You've got these names inverted, the callbacks should be `handle-` the props should be `on-`

Reply:
typescript was going nuts when i passed in onMouseOver

Reply:
Right, so that's a prop that already exists on the element we're extending. It's freaking out because we're altering it's signature. What you'll want to do in this case is just use the `Omit` utility type to remove it from what we're extending we're already omiting onClick, for instance